xwiki-platform-core/xwiki-platform-wysiwyg/xwiki-platform-wysiwyg-api/src/main/java/org/xwiki/wysiwyg/internal/converter/DefaultRequestParameterConverter.java

```diff
@@ -0,0 +1,223 @@
+
+/**
+ * Default implementation of {@link RequestParameterConverter} that handles HTML conversion of parameters needed such
+ * conversion.
+ *
+ * @version $Id$
+ * @since 13.5RC1
+ */
+@Component
+@Singleton
+public class DefaultRequestParameterConverter implements RequestParameterConverter
+{
+    /**
+     * The name of the request parameter whose multiple values indicate the request parameters that require HTML
+     * conversion. For instance, if this parameter's value is {@code [description, content]} then the request has two
+     * parameters, {@code description} and {@code content}, requiring HTML conversion. The syntax these parameters must
+     * be converted to is found also on the request, under {@code description_syntax} and {@code content_syntax}
+     * parameters.
+     */
+    private static final String REQUIRES_HTML_CONVERSION = "RequiresHTMLConversion";
+
+    /**
+     * The name of the session attribute holding the conversion output. The conversion output is stored in a {@link Map}
+     * of {@link Map}s. The first key identifies the request and the second key is the name of the request parameter
+     * that required HTML conversion.
+     */
+    private static final String CONVERSION_OUTPUT = "com.xpn.xwiki.wysiwyg.server.converter.output";
+
+    /**
+     * The name of the session attribute holding the conversion exceptions. The conversion exceptions are stored in a
+     * {@link Map} of {@link Map}s. The first key identifies the request and the second key is the name of the request
+     * parameter that required HTML conversion.
+     */
+    private static final String CONVERSION_ERRORS = "com.xpn.xwiki.wysiwyg.server.converter.errors";
+
+    @Inject
+    private MutableServletRequestFactory mutableServletRequestFactory;
+
+    @Inject
+    private HTMLConverter htmlConverter;
+
+    @Inject
+    @Named("context")
+    private ComponentManager componentManager;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Optional<ServletRequest> convert(ServletRequest request, ServletResponse response) throws IOException
+    {
+        Optional<ServletRequest> result = Optional.of(request);
+        // Take the list of request parameters that require HTML conversion.
+        String[] parametersRequiringHTMLConversion = request.getParameterValues(REQUIRES_HTML_CONVERSION);
+        if (parametersRequiringHTMLConversion != null) {
+            // Wrap the current request in order to be able to change request parameters.
+            MutableServletRequest mreq = this.mutableServletRequestFactory.newInstance(request);
+            // Remove the list of request parameters that require HTML conversion to avoid recurrency.
+            mreq.removeParameter(REQUIRES_HTML_CONVERSION);
+            // Try to convert each parameter from the list and save caught exceptions.
+            Map<String, Throwable> errors = new HashMap<String, Throwable>();
```

Comment:
Can also be slightly simplified by removing the right hand side generic type declarations.
```suggestion
            Map<String, Throwable> errors = new HashMap<>();
```